python/ray/tune/ray_trial_executor.py

```diff
@@ -134,21 +135,18 @@ def _train(self, trial):
 
         self._running[remote] = trial
 
-    def _start_trial(self, trial, checkpoint=None):
+    def _start_trial(self, trial, checkpoint=None, runner=None):
         """Starts trial and restores last result if trial was paused.
 
         Raises:
             RuntimeError if restoring from checkpoint fails.
         """
         prior_status = trial.status
         self.set_status(trial, Trial.RUNNING)
-        trial.runner = self._setup_remote_runner(
+        trial.runner = runner or self._setup_remote_runner(
```

Comment:
Maybe also add a note that we can pass the `cached_actor` through this codepath too?